### PR TITLE
Animate cells

### PR DESCRIPTION
**Pull request by aavina**
Plays a simple animation for cells inheriting the new base adapter class.

### Animation base class
`ChicagoRobotoAdapter` now is used as a base class for `SpeakerAdapter` and `SessionAdapter` which gives the feature to play an animation for views the child adapters show.

They can individually configure the duration or whether or not to play the animation.

`staggerAnimation:Boolean` - Staggers the animations of cells, defaults to `true`
`animationDuration:Long` - The duration of the animation, default to `1000` milliseconds
`showAnimation:Boolean` - Whether to show animation or not, default to `true`

### Kotlin and build tools update
- Updated Kotlin version to `1.2.31`
- Updated Gradle tools to `3.1.1`

### CODE REVIEW

File: android/app/src/main/kotlin/com/chicagoroboto/utils/ChicagoRobotoAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import android.view.ViewGroup
+import android.view.animation.AnimationUtils
+
+/**
+ * Adapter base class to optionally add animations to the view items
+ */
+open class ChicagoRobotoAdapter<T: RecyclerView.ViewHolder>(private var mLastPosition:Int = -1,
+                                                            var mAnimationType:Int = android.R.anim.slide_in_left,
+                                                            var mAnimationDuration:Long = ANIMATION_DURATION,
+                                                            var mShowAnimation:Boolean = true) :
```
Comment (JakeWharton) [inline android/app/src/main/kotlin/com/chicagoroboto/utils/ChicagoRobotoAdapter.kt]: Public fields shouldn't hungariant notation. Moreover, no fields _ever_ should use hungarian notation.

File: android/app/src/main/kotlin/com/chicagoroboto/utils/ChicagoRobotoAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import android.support.v7.widget.RecyclerView
+import android.view.View
+import android.view.ViewGroup
+import android.view.animation.AnimationUtils
+
+/**
+ * Adapter base class to optionally add animations to the view items
+ */
+open class ChicagoRobotoAdapter<T: RecyclerView.ViewHolder>(private var mLastPosition:Int = -1,
```
Comment (JakeWharton) [inline android/app/src/main/kotlin/com/chicagoroboto/utils/ChicagoRobotoAdapter.kt]: `abstract`?

File: android/app/src/main/kotlin/com/chicagoroboto/utils/ChicagoRobotoAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    companion object {
+        const val ANIMATION_DURATION: Long = 1000
+    }
+
+    /**
+     * These won't be used, but will be implemented by child adapter classes.
+     * There's currently no worry on the TODOs getting hit at runtime but if children ever call it
+     * the app can crash.
+     */
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): T {
```
Comment (JakeWharton) [inline android/app/src/main/kotlin/com/chicagoroboto/utils/ChicagoRobotoAdapter.kt]: Delete both these methods once the type is `abstract`

File: android/app/src/main/kotlin/com/chicagoroboto/utils/ChicagoRobotoAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    override fun getItemCount(): Int {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    /**
+     * Play the animation
+     */
+    override fun onBindViewHolder(holder: T, position: Int) {
```
Comment (JakeWharton) [inline android/app/src/main/kotlin/com/chicagoroboto/utils/ChicagoRobotoAdapter.kt]: Add `@CallSuper`

File: android/app/src/main/kotlin/com/chicagoroboto/utils/ChicagoRobotoAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    /**
+     * Shows an animation on the cell
+     * Reference: https://stackoverflow.com/questions/26724964/how-to-animate-recyclerview-items-when-they-appear
+     */
+    private fun setAnimation(view: View, position: Int) {
+        if(position > mLastPosition) {
+            val animation = AnimationUtils.loadAnimation(view.context, mAnimationType)
+            animation.duration = mAnimationDuration
+            view.startAnimation(animation)
```
Comment (JakeWharton) [inline android/app/src/main/kotlin/com/chicagoroboto/utils/ChicagoRobotoAdapter.kt]: Why use view animation which is effectively deprecated over animators?

**Comment by aavina** [comment]
@JakeWharton Addressed everything, thanks for the feedback so far!

**Comment by aavina** [comment]
Introduced some bugs in the last commit, gonna fix them

**Comment by aavina** [comment]
Fixed the bug, no more accidentally hidden/removed events

**Pull request closed without merging**